src/main.cpp

```diff
@@ -1614,22 +1624,40 @@ void readBitmapData(WiFiClient &client)
         // Z1
         if (header == 0x315A)
         {
-          pixel_color = safe_read(client);
+          pixel_color = safe_read_error(client,&read_error);
+          if (read_error){
```

Comment:
Please follow the current coding style for ifs
```c
if (condition)
{
}
```
Once you're going to use the print functionality multiple times, create a static function rather.